tests/integration/test_moveaxis.py

```diff
@@ -44,6 +44,82 @@ def test_moveaxis(ndim, axes):
     assert cn_a.sum() == 0
 
 
+def test_moveaxis_with_empty_axis():
+    np_a = np.ones((3, 4, 5))
+    cn_a = cn.ones((3, 4, 5))
+
+    axes = ([], [])
+    source, destination = axes
+
+    np_res = np.moveaxis(np_a, source, destination)
+    cn_res = cn.moveaxis(cn_a, source, destination)
+    assert np.array_equal(np_res, cn_res)
+
+
+EMPTY_ARRAYS = [
+    [],
+    [[]],
+    [[], []],
+]
+
+
+@pytest.mark.parametrize("a", EMPTY_ARRAYS)
+def test_moveaxis_with_empty_array(a):
+    axes = (0, -1)
+    source, destination = axes
+
+    np_res = np.moveaxis(a, source, destination)
+    cn_res = cn.moveaxis(a, source, destination)
+    assert np.array_equal(np_res, cn_res)
+
+
+class TestMoveAxisErrors:
+    def setup(self):
+        self.x = cn.ones((3, 4, 5))
+
+    def test_repeated_axis(self):
+        msg = "repeated axis"
+        with pytest.raises(ValueError, match=msg):
+            cn.moveaxis(self.x, [0, 0], [1, 0])
+
+        with pytest.raises(ValueError, match=msg):
+            cn.moveaxis(self.x, [0, 1], [0, -3])
+
+    def test_axis_out_of_bound(self):
+        msg = "out of bound"
+        with pytest.raises(np.AxisError, match=msg):
+            cn.moveaxis(self.x, [0, 3], [0, 1])
+
+        with pytest.raises(np.AxisError, match=msg):
+            cn.moveaxis(self.x, [0, 1], [0, -4])
+
+        with pytest.raises(np.AxisError, match=msg):
+            cn.moveaxis(self.x, 4, 0)
+
+        with pytest.raises(np.AxisError, match=msg):
+            cn.moveaxis(self.x, 0, -4)
+
+    def test_axis_with_different_length(self):
+        msg = "arguments must have the same number of elements"
+        with pytest.raises(ValueError, match=msg):
+            cn.moveaxis(self.x, [0], [1, 0])
+
+    def test_axis_with_bad_type(self):
+        msg = "integer argument expected, got float"
+        with pytest.raises(TypeError, match=msg):
+            cn.moveaxis(self.x, [0.0, 1], [1, 0])
+
+        with pytest.raises(TypeError, match=msg):
+            cn.moveaxis(self.x, [0, 1], [1, 0.0])
+
+        msg = "'NoneType' object is not iterable"
+        with pytest.raises(TypeError, match=msg):
+            cn.moveaxis(self.x, None, 0)
+
+        with pytest.raises(TypeError, match=msg):
+            cn.moveaxis(self.x, 0, None)
```

Comment:
split into separate tests for  float and None

Reply:
Fixed